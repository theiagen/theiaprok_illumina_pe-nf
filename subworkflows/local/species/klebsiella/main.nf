include { KLEBORATE } from '../../../../modules/local/species_typing/klebsiella/kleborate/main'

workflow KLEBSIELLA_TYPING {

    take:
    ch_klebsiella // Channel of tuples [meta, assembly]

    main:
    ch_versions = Channel.empty()
    ch_kleborate_results = Channel.empty()

    KLEBORATE (ch_klebsiella)
    ch_kleborate_results = KLEBORATE.out.kleborate_report
    ch_versions = ch_versions.mix(KLEBORATE.out.versions)

    emit:
    kleborate_results = ch_kleborate_results
    versions = ch_versions
}