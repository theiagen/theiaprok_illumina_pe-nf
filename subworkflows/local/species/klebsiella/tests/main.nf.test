nextflow_workflow {

    name "Test Subworkflow KLEBSIELLA_TYPING"
    script "../main.nf"
    workflow "KLEBSIELLA_TYPING"

    test("klebsiella_typing - characterization") {
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test_klebsiella', single_end:false ],
                    file("https://raw.githubusercontent.com/theiagen/test-datasets/refs/heads/kleborate_typing/data/302749_filtered_contigs.fasta"),
                    "Klebsiella pneumoniae"
                ])
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.kleborate_results,
                    workflow.out.versions
                ).match() }
            )
        }
    }
}