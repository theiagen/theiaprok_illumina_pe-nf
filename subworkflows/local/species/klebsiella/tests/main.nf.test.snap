{
    "klebsiella_typing - characterization": {
        "content": [
            [
                [
                    {
                        "id": "test_klebsiella",
                        "single_end": false
                    },
                    "test_klebsiella_kleborate_out.tsv:md5,fa73bd1260fc240b5b4209d60462b6d9"
                ]
            ],
            [
                "versions.yml:md5,d45d4caf711ddae1eebf72604e316a3b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-07-02T21:00:35.257397066"
    }
}